File: solutions/python/difference-of-squares/2/difference_of_squares.py
def square_of_sum(number):
    return sum([item for item in range(1, number + 1)]) ** 2


def sum_of_squares(number):
    return sum([item ** 2 for item in range(1, number + 1)])


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: solutions/c/difference-of-squares/2/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int n) {
    unsigned int sum_squares = 0;
    
    for (unsigned int i = 1; i <= n; i++ ) {
        sum_squares += i * i;
    }

    return sum_squares;
}

unsigned int square_of_sum(unsigned int n) {
    unsigned int sum = 0;

    for (unsigned int i = 1; i <= n; i++ ) {
        sum += i;
    }
   
    return sum * sum;
}

unsigned int difference_of_squares(unsigned int n) {
    return square_of_sum(n) - sum_of_squares(n);
}
